06.Lesson_ThinkingProperties/ThinkingPropertiesExercise/PopsicleFinance/README.md

```diff
@@ -0,0 +1,29 @@
+(Did not have as much time to work on and organize this one)
+
+# Categories
+
+## Unit tests
+- ❌ Unit test for `deposit` - check that rewards and balances are updated
+    - ❓ Is there any way to check things about properties of `accounts` without making it public or using a harness? Would this be a good use of a `ghost`?
+- ✅ `deposit` increases balance
+- ✅ `withdraw` decreases balance
+- ✅ If user’s token balance increased, then `deposit` was called
+- ✅ If decreased, then `withdraw` was called
+- ❌ If someone deposits and then immediately collects fees, they won't earn any fees
+    - ❓ How can I test this one (assuming it's worth testing)? Their ETH balance will decrease by gas costs, so I can't check that their balance is unchanged. Checking that it's nonincreasing feels like it would miss the case when the fees earned are smaller than the gas costs.
+
+## Variable transitions
+- ❌ If `reward` increased, it was from a call to `deposit` or `withdraw`
+    - Need to set up a ghost to access `reward`
+- ❌ If caller's ETH balance increased, it was from a call to `collectFees()`
+    - ❓ This one is failing. It seems to think that a call to `withdraw` triggers an ETH transfer. See anything wrong with my spec `callersETHBalancesIncreasesOnlyFromCollectFees`?
```

Comment:
Not sure the exact counter-example with no report link but it might be due to a havoc. try using the `--settings -optimisticFallback=true`

Reply:
Ah, it's because `withdraw` calls `burn` from `ERC20.sol` and `burn` has an ETH transfer
```
    function burn(address user, uint amount) internal {
        balances[user] -= amount;        
        total -= amount;
        msg.sender.call{value: amount}("");
    }
```